src/cargomonitor.cpp

```diff
@@ -153,7 +154,7 @@ void AddCargoDelivery(CargoID cargo_type, CompanyID company, uint32 amount, Sour
 	for (const Industry * const *ip = st->industries_near.Begin(); ip != st->industries_near.End(); ip++) {
 		CargoMonitorID num = EncodeCargoIndustryMonitor(company, cargo_type, (*ip)->index);
 		CargoMonitorMap::iterator iter = _cargo_deliveries.find(num);
-		if (iter != _cargo_deliveries.end()) iter->second += amount;
+		if (iter != _cargo_deliveries.end() && (*ip)->index == dest) iter->second += amount;
```

Comment:
Make it an `if != dest continue` at the start of the loop? No need to check or find anything just to skip it.